NSS-Spontaneous-Spoonbills/NSS-Spontaneous-Spoonbills
# Added restrictions for certain models, veiwsets

## Link to Ticket
Autolink format: hashtag and number, ex: #1094
Closes #1 
Closes #2 
Closes #3 
Closes #4 
Closes #5 
Closes #6 
Closes #7 
Closes #9 


## Added configurations/restrictions to:
1. User
1. Employee
1. Department


## Steps to Test
1. Run virtual environment
1. Run python server
1. Navigate to http://127.0.0.1:8000/bangazonapi/
1. Open correlating table /[tablename]
1. Click url: value inside correlating table 
1. Most tickets did not have any restrictions on table
Ex. 
```
{
    "url": "http://127.0.0.1:8000/bangazonapi/employee/1/",
    "First_Name": "Vincent",
    "Last_Name": "Monroe",
    "Is_Supervisor": false,
    "Employee_Is_Active": true,
    "Dept_ID": "http://127.0.0.1:8000/bangazonapi/department/10/",
    "Comp_ID": "http://127.0.0.1:8000/bangazonapi/computer/11/"
}
``` 
1. There are three tables to test
1. User, Employee, & Department should not have a delete option in the single instance

```
git fetch --all
git checkout JS_adding_configs
```

## Impacted Areas in Application

##### Modified
bangazonapi/views/Employee_View.py
bangazonapi/views/Department_View.py
bangazonapi/views/User_View.py



Mentions @username
@MrErin @obscure-shadow @pwmurphy89

Definition of Done
The project must be fully documented. This includes the following:
Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
Every class must be documented with purpose, author, and methods.
Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
The project must be able to run fully, and without errors, on each teammate's system.
Fulfills every requirement.
Every line of code has been peer reviewed.
For projects that require unit testing, core functionality must be identified and have at least one test for each
